hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/recordcache/RecordCacheImpl.java

```diff
@@ -181,9 +182,14 @@ public void add(
         final var queue = getQueue();
         final var firstRecord = transactionRecords.get(0);
         removeExpiredTransactions(queue, firstRecord.transactionRecord().consensusTimestampOrElse(Timestamp.DEFAULT));
-
+        // Sort the transaction records based on nonce, so that the user transaction is always first,
+        // followed by any preceding transactions, followed by any child transactions.
+        final var sortedRecords = transactionRecords.stream()
```

Comment:
Noo, the preceding records order was different. Like nonce 3 was given first when I do `getTxnRecord`

Reply:
@Neeharika-Sompalli I believe `RecordListBuilder` orders the list by consensus time [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/RecordListBuilder.java#L293)---which unfortunately is not the same as increasing `nonce`. 😞 

However, I would lean toward putting the sort [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/NetworkTransactionGetRecordHandler.java#L125) in the query response handler, since _functionally_ that's the only place we need it to be ordered by `nonce`.

Reply:
I see. Thanks @tinker-michaelj ! 👍